health-services/project/src/main/java/org/egov/project/web/controllers/LocationCaptureController.java

```diff
@@ -0,0 +1,130 @@
+import org.springframework.stereotype.Controller;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+/**
+ * Controller for handling requests related to location capture tasks.
+ * Provides endpoints for creating and searching location capture tasks.
+ */
+@Controller
+@RequestMapping("/user-location")
+@Validated
+@Slf4j
+public class LocationCaptureController {
+
+    private final HttpServletRequest httpServletRequest;
+    private final LocationCaptureService locationCaptureService;
+    private final Producer producer;
+    private final ProjectConfiguration projectConfiguration;
+
+    /**
+     * Constructor for injecting dependencies into the LocationCaptureController.
+     *
+     * @param httpServletRequest       The HttpServletRequest to capture request details.
+     * @param locationCaptureService   The service for handling location capture logic.
+     * @param producer                 The producer for sending messages to Kafka topics.
+     * @param projectConfiguration     Configuration properties related to the project.
+     */
+    @Autowired
+    public LocationCaptureController(
+            HttpServletRequest httpServletRequest,
+            LocationCaptureService locationCaptureService,
+            Producer producer,
+            ProjectConfiguration projectConfiguration
+    ) {
+        this.httpServletRequest = httpServletRequest;
+        this.locationCaptureService = locationCaptureService;
+        this.producer = producer;
+        this.projectConfiguration = projectConfiguration;
+    }
+
+    /**
+     * Endpoint for creating location capture tasks in bulk.
+     * Receives a UserActionBulkRequest object, processes it, and sends it to the appropriate Kafka topic.
+     *
+     * @param request The bulk request containing user actions to be created.
+     * @return A ResponseEntity containing the response info with HTTP status ACCEPTED.
+     */
+    @RequestMapping(value = "/v1/_create", method = RequestMethod.POST)
+    public ResponseEntity<ResponseInfo> locationCaptureTaskV1BulkCreatePost(
+            @ApiParam(value = "Create Location Capture LocationCapture.", required = true) @Valid @RequestBody UserActionBulkRequest request) {
+        // Set the API ID in the request info using the current request URI.
+        request.getRequestInfo().setApiId(httpServletRequest.getRequestURI());
+
+        try {
+            // Send the request to the Kafka topic for bulk creation.
+            producer.push(projectConfiguration.getBulkCreateLocationCaptureTaskTopic(), request);
+        } catch (Exception e) {
+            log.error("Error sending message to Kafka", e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(
```

Comment:
error message is missing

Reply:
there isn't a way to send error message, it will be printed in logs using log.error as provided.